### DISCUSSION TITLE
DSM Docker force Update?

### DISCUSSION THREAD

**Discussion by max1mvs** [itzg/docker-minecraft-server#1348] [Q&A]
Being kind of a noob in relation to Docker on the Synology DSM, I was wondering how to force an update for the bedrock server using DSM or something similar? I can't really seem to make it do it by itself by stopping and starting the container…

It might also be me not knowing how soon I can actually expect an update. For instance right now, Minecraft Bedrock has been updated to 1.18.10, and the server is currently running 1.18.0. Having no clue as to whether a new version already do exist for the container to update to, that might also be why I cannot see anything happening. Yet, I'm not sure how to be aware of this.

**Comment by itzg** [comment]
It pulls from https://www.minecraft.net/en-us/download/server/bedrock/ which seems to be 1.18.11.01. Can you restart the container and provide the container logs?

**Reply by max1mvs** [reply]
Sure. Here they are:

[marctv-minecraft-bedrock-server.csv](https://github.com/itzg/docker-minecraft-server/files/8035363/marctv-minecraft-bedrock-server.csv)

**Comment by itzg** [comment]
Those don't seem to be the container logs. It should have looked like the following where I'm looking for the "Downloading Bedrock server version" line.

```
DEBU[0000] Using /data to match uid and gid
DEBU[0000] Resolved UID=0 from match path
DEBU[0000] Resolved GID=0 from match path
Looking up latest version...
Downloading Bedrock server version 1.18.11.01 ...
Starting Bedrock server...
NO LOG FILE! - setting up server logging...
[2022-02-10 03:22:20:430 INFO] Starting Server
[2022-02-10 03:22:20:430 INFO] Version 1.18.11.01
[2022-02-10 03:22:20:430 INFO] Session ID 2524fd6a-89c1-4575-8da6-3b79f46bf98e
[2022-02-10 03:22:20:431 INFO] Level Name: Bedrock level
[2022-02-10 03:22:20:434 INFO] Game mode: 0 Survival
[2022-02-10 03:22:20:434 INFO] Difficulty: 1 EASY
[2022-02-10 03:22:20:501 INFO] opening worlds/Bedrock level/db
[2022-02-10 03:22:21:323 INFO] IPv4 supported, port: 19132
[2022-02-10 03:22:21:323 INFO] IPv6 not supported
[2022-02-10 03:22:21:712 INFO] Server started.
[2022-02-10 03:22:21:740 INFO] IPv4 supported, port: 41299
[2022-02-10 03:22:21:740 INFO] IPv6 not supported
```

**Reply by max1mvs** [reply]
Sorry, but perhaps it is I who is the noob here: I was exporting the logs from the container:

<img width="1213" alt="Screenshot 2022-02-10 at 10 47 36" src="https://user-images.githubusercontent.com/17984541/153381467-7442089f-7f85-4360-81aa-553d48336f33.png">

Is there another way of exporting them?

**Reply by itzg** [reply]
That seems like the right place, so now I'm confused why Synology is not showing the logs from container startup. Can you provide a screenshot of the container configuration?

**Reply by max1mvs** [reply]
Sure!
<img width="746" alt="Screenshot 2022-02-10 at 14 38 48" src="https://user-images.githubusercontent.com/17984541/153419527-2a7d901e-3ce4-43bf-a4c5-fffba71e352c.png">
<img width="746" alt="Screenshot 2022-02-10 at 14 38 46" src="https://user-images.githubusercontent.com/17984541/153419529-fec4e7ac-929a-424c-ab65-73514ec97580.png">
<img width="746" alt="Screenshot 2022-02-10 at 14 38 43" src="https://user-images.githubusercontent.com/17984541/153419530-20d3d182-34c8-499f-9ee5-d6d909fcc611.png">
<img width="746" alt="Screenshot 2022-02-10 at 14 38 40" src="https://user-images.githubusercontent.com/17984541/153419533-bbe0d300-385f-40ce-a30e-f7ac04ec28df.png">
<img width="746" alt="Screenshot 2022-02-10 at 14 38 37" src="https://user-images.githubusercontent.com/17984541/153419535-1dde87e0-69d9-4d9b-880f-a4e61936c6ff.png">

**Reply by max1mvs** [reply]
I manually updated the container (I've kept a backup FYI), and now it loads the latest version of Minecraft. Yet, I am not sure if this solves the automatic update part since I don't see the `Looking up latest version...` part in the log file (not even after a restart):

[marctv-minecraft-bedrock-server.csv](https://github.com/itzg/docker-minecraft-server/files/8042402/marctv-minecraft-bedrock-server.csv)

**Reply by itzg** [reply]
Aha, I think you're running the "wrong" image. You're running one maintained by marctv, but mine is `itzg/minecraft-bedrock-server`.

**Reply by itzg** [reply]
...BTW mine will manage the whitelist and server properties so there's no need to map those in specifically. Just the `/data` volume as a whole.

**Reply by max1mvs** [reply]
What? OMG. 😱🤦‍♂️ I am using your Java container – and I was convinced I also used your bedrock server as well (_totally_ overlooked the name there). That is embarrassing. I apologise for the inconvenience and will shift towards your container because it means that the other does not automatically pull the latest one.

**Reply by max1mvs** [reply]
And thank you very much for your help (also from my sons). A donation has already been sent off for your efforts not only here but in general. Enjoy the coffee! 😄
